internal/database/mysql_test.go

```diff
@@ -38,78 +37,134 @@ func setupDBAndOpenSubject(t *testing.T) *internalDatabase.MySQLDatabase {
 	if err != nil {
 		t.Fatal("Failed to create test database -", err)
 	}
-	if err := probe.Close(); err != nil {
-		log.Fatal("Error closing initial database connection -", err)
-	}
 
 	// Connect to the newly created test database
-	probe, err = sql.Open("mysql", "root:"+password+"@tcp(localhost:3306)/test_db")
+	_, err = probe.Exec("USE test_db")
 	if err != nil {
-		t.Fatal("Failed to open connection to test database -", err)
-	}
-	if err := probe.Ping(); err != nil {
-		t.Fatal("Failed to ping test database connection -", err)
+		log.Fatalln("Failed to use test_db -", err)
 	}
 
-	// Open the custom testing database with initial SQL setup
-	subject := internalDatabase.New("test_db", "root", password)
-	if subject == nil {
-		t.Fatal("Failed to open subject database")
-	}
+	// Open the internal database implementation to test
+	subject = New("test_db", "root", password)
 	if err := subject.Open(); err != nil {
 		t.Fatal("Failed to open subject database -", err)
 	}
+	return subject, probe
+}
+
+func teardown(t *testing.T, subject *MySQLDatabase, probe *sql.DB) {
+	t.Helper()
+
+	err := probe.Close()
+	if err != nil {
+		t.Fatal("Failed to close probe database connection -", err)
+	}
+	err = subject.Close()
+	if err != nil {
+		t.Fatal("Failed to close subject database connection -", err)
+	}
+}
 
-	return subject
+func TestConnections(t *testing.T) {
+	subject, probe := setup(t)
+	if err := subject.underlyingDB.Ping(); err != nil {
+		t.Fatal("Failed to open subject db properly -", err)
+	}
+	if err := probe.Ping(); err != nil {
+		t.Fatal("Failed to open probe db properly -", err)
+	}
+	teardown(t, subject, probe)
```

Comment:
This Ping() and the one after it, are already performed in the teardown function. Is it still necessary to do this?

Reply:
I am really only leaving TestConnections for completeness, given how our setup and teardown functions are currently written, this test is really redundant. We have agreed to leave for now tho.